Mathlib/GroupTheory/Perm/Cycle/Concrete.lean

```diff
@@ -518,10 +513,19 @@ def isoCycle' : { f : Perm α // IsCycle f } ≃ { s : Cycle α // s.Nodup ∧ s
 notation3 (prettyPrint := false) "c["(l", "* => foldr (h t => List.cons h t) List.nil)"]" =>
   Cycle.formPerm (Cycle.ofList l) (Iff.mpr Cycle.nodup_coe_iff (by decide))
 
-unsafe instance repr_perm [Repr α] : Repr (Perm α) :=
-  ⟨fun f _ => repr (Multiset.pmap (fun (g : Perm α) (hg : g.IsCycle) => isoCycle ⟨g, hg⟩)
-    (Perm.cycleFactorsFinset f).val -- toCycle is faster?
-    fun _ hg => (mem_cycleFactorsFinset_iff.mp (Finset.mem_def.mpr hg)).left)⟩
-#align equiv.perm.repr_perm Equiv.Perm.repr_perm
+unsafe instance instRepr [Repr α] : Repr (Perm α) where
+  reprPrec f prec :=
+    letI l := Quot.unquot <| Multiset.map repr <| Multiset.pmap toCycle
+      (Perm.cycleFactorsFinset f).val
+      fun _ hg => (mem_cycleFactorsFinset_iff.mp (Finset.mem_def.mpr hg)).left
+    match l with
+    | []  => "1"
+    | [f] => f
+    | l   =>
+      -- multiple terms, use `*` precedence
+      (if prec ≥ 70 then Lean.Format.paren else id)
+      (Lean.Format.fill
+        (Lean.Format.joinSep l (" *" ++ Lean.Format.line)))
```

Comment:
Can you put this part in its own PR? The change to the repr looks good, but I'd prefer to see it with a test, and isolated from all the other computational stuff happening in this PR.

Reply:
#12611